### DISCUSSION TITLE
multicall

### DISCUSSION THREAD

**Discussion by matthewlilley** [wevm/wagmi#418] [Ideas]
Multicall is used heavily in our applications, and I think is needed for specific use cases. Right now we have a mix of wagmi and multicall for contract reads, and the difference in API is a little cumbersome. 

I wonder if there are plans to bring this into wagmi?

**Comment by tmm** [comment]
Definitely interested! @matthewlilley (or others reading this), feel free to use this discussion topic to post requirements or info about how you think this should work (or not work) in wagmi so we can discuss async. Also, would love to chat async too if you drop your contact info.

**Comment by chillichelli** [comment]
For multicall it would make sense to have a config option that allows the user to switch between an all or nothing or resolve independently mode.

E.g.: when loading token balances with a loader spinner, a resolve independently may make more sense UX wise and when the returned data is used in conjunction to derive new data, an all or nothing mode is required. The latter would eliminate the need for certain helper contracts (which are a pain when doing multichain)

**Comment by jxom** [comment]
This is currently being worked on! 

https://github.com/tmm/wagmi/pull/511

**Reply by nasa8x** [reply]
@jxom @tmm 
Officially published yet? I have not seen the updated documentation on the homepage.
Thank you for your contributions.

**Reply by jxom** [reply]
It is nearly there!
